test/ccmath_abs_test.cpp

```diff
@@ -0,0 +1,88 @@
+//  (C) Copyright Matt Borland 2021.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+#include <boost/math/ccmath/abs.hpp>
+#include <boost/math/ccmath/fabs.hpp>
+#include <boost/math/tools/config.hpp>
+
+#ifdef BOOST_HAS_FLOAT128
+#include <boost/multiprecision/float128.hpp>
+#endif
+
+template <typename T>
+void test()
+{
+    static_assert(boost::math::ccmath::abs(T(3)) == 3);
+    static_assert(boost::math::ccmath::abs(T(-3)) == 3);
+    static_assert(boost::math::ccmath::abs(T(-0)) == 0);
+    static_assert(boost::math::ccmath::abs(-std::numeric_limits<T>::infinity()) == std::numeric_limits<T>::infinity());
```

Comment:
Could some very simple expressions be joined here? eg

```
static_assert(abs(sin(1)) == ...)
```

Reply:
The difficulty is the only functions guaranteed to be `constexpr` are `pow` and `sqrt`. I could do something heavily restricted like only testing your suggestion on GCC10+ where some of `cmath` is `constexpr`. Thoughts?

Reply:
Yeah, I think if you have a guarantee like that, then you should *try* to test it. If it dies in CI, well, not a big deal.